bigtable/beam/bulk-data-generator/src/main/java/bigtable/BulkWrite.java

```diff
@@ -0,0 +1,201 @@
+          .format("Create mutations that write %d MB to each row", MB_PER_ROW);
+
+      System.out.println(generateLabel);
+      System.out.println(mutationLabel);
+
+      PCollection<Mutation> mutations = p
+          .apply(generateLabel, GenerateSequence.from(0).to(numRows)
+              .withRate(rate, Duration.standardSeconds(1)))
+          .apply(mutationLabel,
+              ParDo.of(new CreateMutationFn()));
+
+      for (String tableId : newTableIds) {
+        mutations.apply(String.format("Write data to table %s", tableId),
+            CloudBigtableIO.writeToTable(new CloudBigtableTableConfiguration.Builder()
+                .withProjectId(options.getProject())
+                .withInstanceId(options.getBigtableInstanceId())
+                .withTableId(tableId)
+                .build()));
+      }
+
+      p.run();
+    } catch (IOException e) {
+      System.out.println("Unable to connect to Bigtable instance.");
+    }
+  }
+
+  private static List<String> getNewTableIds(BigtableTableAdminClient adminClient,
+      double expectedSize) {
+    List<String> tableIds = adminClient.listTables();
+    List<String> newTableIds = new ArrayList<>();
+    double currentSize = tableIds.size() * TB_PER_TABLE;
+
+    if (currentSize >= expectedSize) {
+      int numTablesToDelete = (int) ((currentSize - expectedSize) / .5);
+      for (int i = 0; i < numTablesToDelete; i++) {
+        adminClient.deleteTable(tableIds.get(i));
+      }
+      System.out.printf("Deleted %d tables%n", numTablesToDelete);
+    } else {
+
+      int numTablesToCreate = (int) ((expectedSize - currentSize) / .5);
+      System.out.printf("Creating %d tables%n", numTablesToCreate);
+
+      for (int i = 0; i < numTablesToCreate; i++) {
+        String tableId = TABLE_PREFIX + UUID.randomUUID().toString().substring(0, 20);
+        CreateTableRequest createTableRequest = CreateTableRequest.of(tableId)
+            .addFamily(COLUMN_FAMILY);
+        adminClient.createTable(createTableRequest);
+        newTableIds.add(tableId);
+
+        System.out.println(tableId);
+      }
+    }
+
+    return newTableIds;
+  }
+
+  private static int getClusterNodeCount(String projectId, String instanceId) {
+    try {
+      BigtableClusterUtilities clusterUtility = BigtableClusterUtilities
+          .forInstance(projectId, instanceId);
+      Cluster cluster = clusterUtility.getSingleCluster();
+      String clusterId = new BigtableClusterName(cluster.getName()).getClusterId();
+      String zoneId = BigtableClusterUtilities.getZoneId(cluster);
+      int clusterNodeCount = clusterUtility.getClusterNodeCount(clusterId, zoneId);
+      System.out.println("Cluster size " + clusterNodeCount);
+      return clusterNodeCount;
+    } catch (IOException | GeneralSecurityException e) {
+      System.out.println("Unable to get cluster size. Treating as single-node cluster.");
+      return 1;
+    }
+  }
+
+  static class CreateMutationFn extends DoFn<Long, Mutation> {
```

Comment:
NIT: I would add a comment here that describes why/how you need to create this subclass of `DoFn`.

Reply:
this is a pretty standard apache beam pattern

Reply:
Okay